src/nvim/testdir/test_search.vim

```diff
@@ -723,6 +723,30 @@ func Test_incsearch_substitute_dump()
   call delete('Xis_subst_script')
 endfunc
 
+func Test_incsearch_highlighting()
```

Comment:
Port this toa lua screen test but keep this for easier merging.

Reply:
I recycled the test from https://github.com/neovim/neovim/pull/14215 - funnily enough while the vim test didn't catch the missing updates in screen.c my test did.